tap_marketo/sync.py

```diff
@@ -184,14 +184,25 @@ def get_or_create_export_for_activities(client, state, stream, export_start, con
         # that is not a real field. `createdAt` proxies `activityDate`.
         # The activity type id must also be included in the query. The
         # largest date range that can be used for activities is 30 days.
-        export_end = get_export_end(export_start, end_days=int(config.get('max_export_days', MAX_EXPORT_DAYS)))
+        max_export_days = int(config.get('max_export_days',
+                                         MAX_EXPORT_DAYS))
+        export_end = get_export_end(export_start,
+                                    end_days=max_export_days)
         query = {"createdAt": {"startAt": export_start.isoformat(),
                                "endAt": export_end.isoformat()},
                  "activityTypeIds": [activity_type_id]}
 
         # Create the new export and store the id and end date in state.
         # Does not start the export (must POST to the "enqueue" endpoint).
-        export_id = client.create_export("activities", ACTIVITY_FIELDS, query)
+        try:
+            export_id = client.create_export("activities", ACTIVITY_FIELDS, query)
+        except client.ApiException as e:
+            raise client.ApiException(
+                ("You may wish to consider changing the "
+                 "`max_export_days` config value to a lower number if "
+                 "you're unable to sync a single {} day window within "
+                 "your current API quota.").format(
+                     max_export_days)) from e
```

Comment:
Looks like the client can throw this in a few different situations, not just API limit exceeded. It would probably be better to add a `QuotaExceededException` from the client when appropriate and handle that for this message.